SpiceSharp/Circuits/CircuitState.cs

```diff
@@ -126,33 +176,45 @@ public enum DomainTypes
         /// Get the states for this circuit
         /// Each element in the vector is used by circuit components to store their state
         /// </summary>
-        public Vector<double>[] States { get; private set; } = null;
+        public double[][] States { get; private set; } = null;
 
         /// <summary>
         /// True if already initialized
         /// </summary>
         public bool Initialized { get; private set; } = false;
 
         /// <summary>
+        /// Constructor
+        /// </summary>
+        public CircuitState()
+        {
+            Matrix = new Matrix();
+        }
+
+        /// <summary>
         /// Initialize the state
         /// </summary>
         /// <param name="ckt"></param>
         public void Initialize(Circuit ckt)
         {
             // Initialize all matrices
             Order = ckt.Nodes.Count + 1;
-            Real = new CircuitStateReal(Order);
-            Complex = new CircuitStateComplex(Order);
+            Rhs = new double[Order];
+            iRhs = new double[Order];
+            Solution = new double[Order];
+            OldSolution = new double[Order];
+            iSolution = new double[Order];
+
             Noise = new CircuitStateNoise();
             if (ckt.Method != null)
                 ReinitStates(ckt.Method);
             else
-                States = new Vector<double>[2];
+                States = new double[2][];
```

Comment:
The first index indicates the history of the state. States[0] is the current state, States[1] is the state at the previous timepoint, etc.
Integration methods need to look back in time for a number of points. If there is no integration method (which is the case if the solution should have no time-dependency), then no history is necessary and the current state is enough.
At least, that's the theory. In practice MOS1-2-3 use States[1] to calculate "Meyer's capacitors" when calculating small-signal parameters. Hence 2 instead of 1.

I believe a better solution would be adapting the models such that we can change this to 1. But maybe this is not the place to do that.